File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
const climbStairs = n => {
    const stairArr = new Array(n + 1);
    stairArr[0] = 0;
    stairArr[1] = 1;
    stairArr[2] = 2;
    for(let i = 3; i < n + 2; i++) {
        stairArr[i] = stairArr[i - 2] + stairArr[i - 1];
    }
    return stairArr[n];
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 1:
            return 1
        arr = [None] * n
        arr[0] = 1
        arr[1] = 2
        i = 2
        while i < n:
            arr[i] = arr[i - 1] + arr[i - 2]
            i += 1
        print(arr)
        return arr[-1]